components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/factory/HttpUnRegistrationResponseFactoryTest.java

```diff
@@ -0,0 +1,212 @@
+import static org.mockito.Matchers.eq;
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@PrepareForTest({HttpUnregistrationResponseFactory.class, HttpIdentityResponse.HttpIdentityResponseBuilder.class})
+public class HttpUnRegistrationResponseFactoryTest extends PowerMockTestCase {
+
+    private UnregistrationResponse mockUnregistrationResponse;
+    private HttpUnregistrationResponseFactory httpUnregistrationResponseFactory;
+    private HttpIdentityResponse.HttpIdentityResponseBuilder mockHttpIdentityResponseBuilder;
+
+    @BeforeMethod
+    private void setUp() {
+
+        mockUnregistrationResponse = mock(UnregistrationResponse.class);
+        httpUnregistrationResponseFactory = new HttpUnregistrationResponseFactory();
+
+    }
+
+    @DataProvider(name = "instanceProvider")
+    public Object[][] getInstanceType() {
+
+        mockUnregistrationResponse = mock(UnregistrationResponse.class);
+        IdentityResponse identityResponse = mock(IdentityResponse.class);
+        return new Object[][]{
+                {mockUnregistrationResponse, true},
+                {identityResponse, false}
+        };
+    }
+
+    @Test(dataProvider = "instanceProvider")
+    public void testCanHandle(Object identityResponse, Boolean expected) throws Exception {
+
+        if (expected) {
+            assertTrue(httpUnregistrationResponseFactory.canHandle((UnregistrationResponse) identityResponse));
+        } else {
+            assertFalse(httpUnregistrationResponseFactory.canHandle((IdentityResponse) identityResponse));
+        }
+    }
+
+    @DataProvider(name = "exceptionInstanceProvider")
+    public Object[][] getExceptionInstanceType() {
+
+        FrameworkException exception1 = new UnRegistrationException("");
+        FrameworkException exception2 = new UnRegistrationException("", "dummyMessage");
+        FrameworkException exception3 = new UnRegistrationException("", new Throwable());
+        FrameworkException exception4 = new UnRegistrationException("", "dummyMessage", new Throwable());
+        FrameworkException exception5 = new FrameworkException("");
+        return new Object[][]{
+                {exception1, true},
+                {exception2, true},
+                {exception3, true},
+                {exception4, true},
+                {exception5, false}
+        };
+    }
+
+    @Test(dataProvider = "exceptionInstanceProvider")
+    public void testCanHandleException(Object exception, boolean expected) throws Exception {
+
+        if (expected) {
+            Assert.assertTrue(httpUnregistrationResponseFactory.canHandle((UnRegistrationException) exception));
+        } else {
```

Comment:
I think this also will be more readable if we have the unhappy case as a separate test method